benches/benches/l1_database.rs

```diff
@@ -0,0 +1,275 @@
+//! Benchmarks for [`L1Database`] trait implementations.
+//!
+//! This module benchmarks the core operations of the [`L1Database`] trait using `RocksDB`
+//! as the storage backend. The benchmarks test various data sizes and operation patterns
+//! to measure performance characteristics.
+
+use std::hint::black_box;
+
+use alpen_benchmarks::db::{
+    create_temp_rocksdb, default_db_ops_config, l1_data_gen::L1DataGenerator,
+};
+// Suppress unused crate warnings
+#[allow(unused_imports)]
+use bitcoin as _;
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+#[allow(unused_imports)]
+use rockbound as _;
+use strata_db::traits::L1Database;
+use strata_db_store_rocksdb::l1::db::L1Db;
+#[allow(unused_imports)]
+use strata_primitives as _;
+#[allow(unused_imports)]
+use strata_state as _;
+use tempfile::TempDir;
+
+/// Transaction counts to test across benchmarks.
+const TX_COUNTS: &[usize] = &[1, 2, 3, 5, 10, 20, 50];
+
+/// Block counts to test for chain operations.
+const BLOCK_COUNTS: &[usize] = &[1, 2, 3, 5, 10, 20, 50];
```

Comment:
Do you think we should have counts at least in the order of 1000 to actually test db under load?

Reply:
My initial implementation had 100, 250 and 1_000 but I thought it was overkill. I'm reconsidering now. Will extend this range to thousands.